### ISSUE TITLE
[Issue]: No Album Artists leads to all artists being displayed

### ISSUE THREAD

**Issue by twihno** [jellyfin/jellyfin#10052]
### TL;DR
No album artist leads to undesired behavior

### Long version
When an album has an album artist, but the individual artists vary, jellyfin displays the artists under every track and shows the album artist in the header.

Now the problem with compilations: They don't (normally) have an album artist.
Sometimes, "Various artists" (or something like that) gets written into the tags and that works with jellyfin.
But theoretically this is wrong: There is no album artist (and various artists also gets their own page even though they don't exist)
(This is actually the old behavior of Spotify).

#### What does Jellyfin actually do, when no album artist is available?

Every single artist gets credited as album artist in the header bar (and causes massive overflowing) and the individual tracks don't show their respective artist.
The "album artists" also get sorted alphabetically in jellyfin-web (this is desired behavior) which doesn't help with guessing which artist actually corresponds to which song.

Jellyfin web:
The indidual tracks still have the context menu entry "Show album artist" which shows the artist of the corresponding track (this button shouldn't be available in this case)

Other clients like e.g. Finamp show the individual artists, but the album artist is still wrong. In this case it's only the first one in the alphabet.
So I GUESS this behavior comes from the backend. (Please correct me if I'm wrong)

#### So what is the new behavior of Spotify?
Instead of showing the "fake artist" "Various Artist", the UI shows the greyed-out text "Various Artist" translated to the current language.

#### What should Jellyfin do?
(My suggestions)

- Maybe tag these albums (no album artist, multiple artists) as compilations or something like that? Perhaps clients can handle this specific case more easily with this information
- Show "Various Artists" (best case: translated) as album artist without generating an artist page. Clients should not make this text clickable. Maybe an option would be to leave the album artist field empty (I don't know what the current behavior is) and let the clients handle this case. Hot take: every piece of music/media was created by at least one person. So if no person is credited for the whole unit/album showing "Various Artists" would always be correct because multiple artists worked on it?
_(I think this would be the best and easiest option)_
- Jellyfin-web: Show the individual artist per Track

### Jellyfin Version

10.8.z

### Environment

```markdown
- Virtualization: Docker
- Clients: Jellyfin-Web, Finamp
```

### Please attach any screenshots here

![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/ca19eba0-51f9-434d-8f01-38d59b9796de)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by HeadBoffin** [comment]
Just setup my first Jellyfin on a Pi 4 - it rocks, it is as awesome as Jeff Geerling demonstrated on YT, overall very impressed.

But I found this issue after the surprise of importing my carefully tagged library from an old MacBook running iTunes. I personally like to differentiate between a track being on a compilation album, like the above, and being on an artist/bands own album. I've learn't to live with all the random tracks on complications being shown on an artists page.

Having all the artists in the header ends up being a hot mess - overflowing the area when there's a fair number of them &/or longer names. Not having the artist against each track is unexpected (that's British understatement).

Happy to help. I've had a peek in library.db,  if someone can show me the architectural docs for the web interface plus where the server pulls in meta data I can have a poke around.

**Comment by Xitee1** [comment]
I have a similar issue but it's much more intense (it's more of a big playlist and not an album).
The UI is pretty much unusable unless you scroll down.
![chrome_GQmoibzQ0H](https://github.com/jellyfin/jellyfin/assets/59659167/9d7dc783-0196-4817-abb6-0e809179d0e5)

**Comment by elliottoille** [comment]
I also have this same problem, really annoys me.  I also don't want all of these artists to show up under the "album artists page", I don't want 100 entries with 1 single song on a compilation album each.
![screenshot-20240201-214414Z-current](https://github.com/jellyfin/jellyfin/assets/64473356/eece5aba-74a7-487e-a72f-dae5e9c065e2)
seems like it should be a simple enough fix, but who am I to judge

**Comment by felix920506** [comment]
Closing as duplicate of https://github.com/jellyfin/jellyfin-web/issues/4228 but will carry over image from @Xitee1 's comment to show severity

**Comment by twihno** [comment]
I played around with the API and I think that this is a backend issue:

(all api calls with parentId set; irrelevant parts were removed)

# Good example

![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/484fa482-6134-4a0e-9288-c7b14594d3c5)

## `/Artists/AlbumArtists` endpoint:
```[json]
{
	"Items": [
		{
			"Name": "Lena Raine",
                        /* not relevant */
		}
	],
	"TotalRecordCount": 1,
	"StartIndex": 0
}
```
## `/Artists` endpoint
```[json]
{
	"Items": [
		{
			"Name": "Lena Raine",
                        /* not relevant */
		},
		{
			"Name": "Tanioka Kumi",
                        /* not relevant */
		}
	],
	"TotalRecordCount": 2,
	"StartIndex": 0
}
```

# Bad example

![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/2b35be2d-a477-4567-b072-c9364e75c7ef)

## `/Artists/AlbumArtists` endpoint
```[json]
{
  "Items": [
    {
      "Name": "10cc"
      /* not relevant */
    },
    {
      "Name": "Blue Swede"
      /* not relevant */
    },
    {
      "Name": "David Bowie"
      /* not relevant */
    },
    {
      "Name": "Elvin Bishop"
      /* not relevant */
    },
    {
      "Name": "The Five Stairsteps"
      /* not relevant */
    },
    {
      "Name": "Jackson 5"
      /* not relevant */
    },
    {
      "Name": "Marvin Gaye & Tammi Terrell"
      /* not relevant */
    },
    {
      "Name": "Norman Greenbaum"
      /* not relevant */
    },
    {
      "Name": "The Raspberries"
      /* not relevant */
    },
    {
      "Name": "Redbone"
      /* not relevant */
    },
    {
      "Name": "The Runaways"
      /* not relevant */
    },
    {
      "Name": "Rupert Holmes"
      /* not relevant */
    }
  ],
  "TotalRecordCount": 12,
  "StartIndex": 0
}
```

## `/Artists` endpoint
```[json]
{
  "Items": [
    {
      "Name": "10cc"
      /* not relevant */
    },
    {
      "Name": "Blue Swede"
      /* not relevant */
    },
    {
      "Name": "David Bowie"
      /* not relevant */
    },
    {
      "Name": "Elvin Bishop"
      /* not relevant */
    },
    {
      "Name": "The Five Stairsteps"
      /* not relevant */
    },
    {
      "Name": "Jackson 5"
      /* not relevant */
    },
    {
      "Name": "Marvin Gaye & Tammi Terrell"
      /* not relevant */
    },
    {
      "Name": "Norman Greenbaum"
      /* not relevant */
    },
    {
      "Name": "The Raspberries"
      /* not relevant */
    },
    {
      "Name": "Redbone"
      /* not relevant */
    },
    {
      "Name": "The Runaways"
      /* not relevant */
    },
    {
      "Name": "Rupert Holmes"
      /* not relevant */
    }
  ],
  "TotalRecordCount": 12,
  "StartIndex": 0
}

```

The metadata embedded in the file (and therefore used in the import):
![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/f7a85863-a784-4619-b49b-6251916838d5)

The metadata in jellyfin:
![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/ab865156-a861-4e26-b662-49bc9f7cd2f1)
(it shows an album artist(?))

I deleted the album artist in the first two entries (in jellyfin-web) and this is what happened after a browser reload:
![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/b9fb2b9f-d7be-46bb-9369-19133894bd5e)
(It now shows the artist next to each track. Yay!)

I know manually deleted all album artists in the album. The UI stays the same but the api looks quite different:

`/Artists/AlbumArtists` endpoint
```[JSON]
{
	"Items": [],
	"TotalRecordCount": 0,
	"StartIndex": 0
}
```

That's how it is supposed to be (at least the API result, not the UI)
The `/Artists` endpoint stayed the same.

Finamp still shows the alphabetically first artist as the album artist

# So what are the problems (in my opinion)
1. the import doesn't work and jellyfin somehow copies the artist to the album artist
2. the ui (jellyfin-web, finamp, ...) doesn't know how to handle empty album artists

The first issue is a backend issue and therefore should be investigated further.

The second issue should be solved (in my opinion) by showing a grey/unclickable "Various Artists" text in the UI. This doesn't require any changes to the backend/API. Jellyfin-web also already handles the "show the individual artists per track" aspect correctly. I already talked about my solution in the inital issue:
> Show "Various Artists" (best case: translated) as album artist without generating an artist page. Clients should not make this text clickable. Maybe an option would be to leave the album artist field empty (I don't know what the current behavior is) and let the clients handle this case. Hot take: every piece of music/media was created by at least one person. So if no person is credited for the whole unit/album showing "Various Artists" would always be correct because multiple artists worked on it?
*(I think this would be the best and easiest option)*

I guess/really hope that this is reproducible. At least multiple people (in this issue and also the linked one) have the same symptoms

**Comment by twihno** [comment]
I **GUESS** I also found the code responsible for issue 1

![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/075e61c3-e2d4-4855-9f11-a509d91dbd29)

IMO it isn't right to set the AlbumArtist to the Artist if no AlbumArtist (e.g. compilation) is provided.

Leaving the albumArtist field empty seems to work with the rest of the code (based on the findings in my previous comment) so removing these lines and making the clients handle an empty album artist should solve this and the corresponding issues.

**Comment by twihno** [comment]
Ok, I cloned the repo and launched a clean version of the current main branch (commit 81cf4b6c509fcd2f3669154c051de2a598841dd9). In this version it seems to work (with mp3 and FLAC). Ahhh.

![grafik](https://github.com/jellyfin/jellyfin/assets/82888998/152d8381-6450-47dd-93ef-20a5541dcd84)

The album artist metadata field stays empty and the album artist row isn't shown in the web ui (maybe a little indication would be nice) but whyyyyyy???

Maybe this commit fixed it??????? https://github.com/jellyfin/jellyfin/commit/66ff724acf6bac3e8e65773ee46bb698b2cda334

😭

**Comment by felix920506** [comment]
Hi, this has been closed as a duplicate. Please comment on https://github.com/jellyfin/jellyfin-web/issues/4228
